File: backend/SmakApi/Models/DTOs/RecipeSearchQuery.cs
namespace SmakApi.Models.DTOs;

public class RecipeSearchQuery
{
    public string? Search { get; set; }
    public Guid? CategoryId { get; set; }
    public string? AuthorName { get; set; }
    public Guid? AuthorId { get; set; }
    public int? MaxCookTime { get; set; }
    public double? MinRating { get; set; }
    public bool? OnlyFavorites { get; set; }
    public string? SortBy { get; set; } // "rating", "time", "popular"
}

File: mobile/app/src/main/java/com/itstep/smakmobile/dto/RecipeSearchQuery.java
package com.itstep.smakmobile.dto;

import java.util.UUID;

public class RecipeSearchQuery {
    public String search;
    public UUID categoryId;
    public String authorName;
    public UUID authorId;
    public Integer maxCookTime;
    public Double minRating;
    public Boolean onlyFavorites;
    public String sortBy;
}

